src/plugins/shell/shell.c

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ *
+ * @brief Source for shell plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "shell.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+
+
+int elektraShellOpen (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	// plugin initialization logic
+
+	return 1; // success
+}
+
+int elektraShellClose (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	// free all plugin resources and shut it down
+
+	return 1; // success
+}
+
+static int executeCommand(const char *cmdline)
+{
+	char *cmd[4];
+	cmd[0] = "/bin/sh";
+	cmd[1] = "-c";
+	cmd[2] = cmdline;
+	cmd[3] = NULL;
+	pid_t pid = fork();
+	if(pid == 0)
+	{
+		return execv("/bin/sh", cmd);
```

Comment:
It would be good to have a way to suppress output and, on errors, putting all text into the error reason would be nice.

Not really must-have features though.

Another usecase is to send/receive a keyset on stdin/stdout and use a shell script as filter. But I think this would be a different story.